compiler/rustc_codegen_ssa/src/codegen_attrs.rs

```diff
@@ -779,6 +787,49 @@ fn check_link_name_xor_ordinal(
     }
 }
 
+#[derive(Default)]
+struct MixedExportNameAndNoMangleState {
+    export_name: Option<Span>,
+    hir_id: Option<HirId>,
+    no_mangle: Option<Span>,
+    no_mangle_attr_name: Option<String>,
```

Comment:
We should try store a reference to the attribute instead of pretty-printing it every time.
```suggestion
    no_mangle_attr: Option<&'a ast::Attribute>,
```